ddtrace/context.py

```diff
@@ -52,16 +53,17 @@ def __init__(
         if sampling_priority is not None:
             self._metrics[SAMPLING_PRIORITY_KEY] = sampling_priority
 
-        self._lock = threading.RLock()
+        if lock is not None:
+            self._lock = lock
+        else:
+            self._lock = threading.RLock()
```

Comment:
🤔  I wonder if this pattern creates the lock _before_ evaluating the `or`? I believe it does since Python is not lazy ☹️ , so keep the initial coding, but with the `forksafe` variant?

Reply:
did we not need a `forksafe` here before? is that a bug?

@Kyle-Verhoog ?